app/main/views/manage_users.py

```diff
@@ -59,8 +59,15 @@ def invite_user(service_id, user_id=None):
     if user_id:
         user_to_invite = User.from_id(user_id)
     elif form.email_address.data:
-        user_to_invite = User.from_email_address_or_none(form.email_address.data)
-
+        # Check if user with email address already exists
+        if User.from_email_address_invited(form.email_address.data) is not None:
+            user_to_invite = User.from_email_address_invited(form.email_address.data)
```

Comment:
Saves you calling the API twice?

```suggestion
        already_invited_user = User.from_email_address_invited(form.email_address.data)
        if already_invited_user is not None:
            user_to_invite = already_invited_user
```